tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,319 @@
+    my $get_vm_hostnames   = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames       = script_output($get_vm_hostnames, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
+    foreach (@vm_hostnames_array)
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define $_.xml");
+	upload_logs "$_.xml";
+    }
+
+}
+
+sub restore_network {
+
+    my $network_mark  = "/etc/sysconfig/network/ifcfg-br0.new";
+    if (script_run("[[ -f $network_mark ]]") == 0){
+        assert_script_run("ip link set dev $virt_host_bridge down",60);
+        my $bridge_destroy = <<'EOF';
+#!/bin/bash
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='dhcp'\nSTARTMODE='auto'"
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+rm -rf /etc/sysconfig/network/ifcfg-br0*
+systemctl restart network.service;
+EOF
+         script_output($bridge_destroy, 900);
+    }
+
+}
+
+
+sub run {
+
+     #Prepare VM HOST SERVER Network Interface Configuration
+     #for libvirt virtual network testing 
+     prepare_network;
+
+     # Remove SSH old files and Generate the key pair
+     assert_script_run "rm -rf ~/.ssh/* || true;ssh-keygen -t rsa -P '' -f ~/.ssh/id_rsa";
+ 
+     # Install required packages
+     zypper_call '-t in iptables iputils bind-utils sshpass';
+
+     # Guest SSH Setup
+     my $gi_guest = '';
+     foreach my $guest (keys %xen::guests) {
+	 #Archive deployed Guests
+         assert_script_run("virsh dumpxml $guest > $guest.xml");
+         record_info "$guest", "Establishing SSH connection to $guest";
+         assert_script_run("virsh start $guest", 60);
+         save_screenshot;
+         $gi_guest = script_output("sleep 40;journalctl | grep dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+	 #Copy the VM host SSH Key to guest systems
+         assert_script_run("sshpass -p novell ssh-copy-id -f root\@$gi_guest", 60);
+	 #Prepare the new guest network interface files for libvirt virtual netwok
+         assert_script_run("ssh root\@$gi_guest 'cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth1;cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth2'");
+     }
+
+#NAT BASED NETWORK
+##Prepare Test
+     die "There was not the default(NAT) virtual network" if (script_run('virsh net-list --all | grep default') != 0);
+     
+     foreach my $guest (keys %xen::guests) {
+         record_info "virsh-virtual_network", "NAT BASED NETWORK";
+	 #Define and start NAT BASED NETWORK
+     	 assert_script_run("virsh net-info default");
+     	 assert_script_run("virsh net-dumpxml default |tee vnet_nated.xml");
+     	 assert_script_run("virsh net-undefine default");
+     	 assert_script_run("virsh net-define vnet_nated.xml");
+	 assert_script_run("virsh net-start default");
```

Comment:
Maybe you can give the re-defined network a different name other than "default" ?

Reply:
Good idea, will update this as your mentioned

Reply:
Good idea.